docs/adr/0005-api-for-spaces.md

```diff
@@ -0,0 +1,118 @@
+
+The purpose of this new API is to give clients a very simple way to query the dynamic list of spaces, that the user has access to. Clients can provide a way better user experience with that.
+
+This API would even allow to provide (WebDAV-) endpoints depending on the kind and version of the client asking for it.
+
+This ADR discusses how this API could be designed.
+
+## Decision Drivers
+
+- Make it easy to work with a dynamic list of spaces of a user for the clients.
+- No longer the need to make assumptions about WebDAV- and other routes in clients.
+- More meta data available about spaces for a better user experience.
+- Part of the bigger spaces plan.
+- Important to consider in client migration scenarios, ie. in CERN.
+
+## Considered Options
+
+1. [Microsoft Graph API](https://developer.microsoft.com/en-us/graph) inspired API that provides the requested information.
+
+## Decision Outcome
+
+This the DRAFT for the API.
+
+### API to Get Info about Spaces
+
+ownCloud servers provide an API to query for available spaces of an user.
+
+Most important, the API returns the WebDAV endpoint for each space. With that, clients do not have to make assumptions about WebDAV routes any more.
+
+See [Drive item in Microsoft Graph API](https://docs.microsoft.com/de-de/graph/api/resources/onedrive?view=graph-rest-1.0)
+
+### Get "Home folder"
+
+Retrieve information about the home space of a user. Note: The user has access to more spaces. This call only returns the home space to provide API parity with the Graph API.
+
+API Call: `/me/drive`: Returns the information about the users home folder.
+
+### Get All Spaces of a User
+
+Retrieve a list of available spaces of a user. This includes all spaces the user has access to at that moment, also the home space.
+
+API Call: `/me/drives`: Returns a list of spaces.
+
+### Common Reply
+
+The reply to both calls is either one or a list of [Drive representation objects](https://docs.microsoft.com/de-de/graph/api/resources/drive?view=graph-rest-1.0):
+
+```
+{
+  "id": "string",
+  "createdDateTime": "string (timestamp)",
+  "description": "string",
+  "driveType": "personal | projectSpaces | shares",
+  "driveStatus": "accepted | pending | mandatory",
+  "eTag": "string",
+  "lastModifiedDateTime": "string (timestamp)",
+  "name": "string",
+  "owner": { "@odata.type": "microsoft.graph.identitySet" },
+  "webUrl": "url",
+  "coowner": [ { "@odata.type": "microsoft.graph.identitySet" } ]
+}
+
+```
+
+The meaning of the object are in ownClouds context:
+
+1. **id** - a unique ID identifying the space
+2. **driveType** - describing the type of the space.
+3. **driveStatus** - telling the status
+4. **eTag** - the current ETag of the space
+5. **owner** - an owner object to whom the space belongs
+6. **webUrl** - the relative Webdav path in ownCloud. It must include the `remote.php` part in oC10 installations.
```

Comment:
`must incude` is a bit strong, because you can set up [`index.php`-less URLs](https://doc.owncloud.org/server/10.0/admin_manual/configuration/server/index_php_less_urls.html) and why relative? what about pointing to multiple domains to shard even further?

Reply:
the root item contains a webDavURL property that should be used